src/po/components/components.js

```diff
@@ -42,4 +39,51 @@ export class TrelloComponents {
   addNewCard() {
     return $$("[data-testid='list-add-card-button']")[1];
   }
+
+  getListByTitle(title) {
+    return $(`//h2[@data-testid='list-name' and text()='${title}']`);
+  }
+
+  getCardByName(name) {
+    return $(`//a[@data-testid='card-name' and text()='${name}']`);
+  }
+
+  generateRandomString() {
+    return (Math.random() + 1).toString(36).substring(5);
+  }
+
+  generateTitle(type) {
+    if (type !== "list" && type !== "card" && type !== "board") {
+      throw new Error(`Invalid ${type}. Use 'list' or 'card' or 'board'`);
+    }
+    return `${
+      type.charAt(0).toUpperCase() + type.slice(1)
+    }_${this.generateRandomString()}`;
+  }
+
+  parentDivList() {
+    return $("//div[@data-testid='list']//h2[text()='Tasks']");
+  }
+
+  async getAllCards() {
+    return await this.parentDivList.$$("li[data-testid='list-card']");
+  }
+
+  urgentCards = [];
+  nonUrgentCards = [];
+
+  async checkOnlyUrgentCardsDisplayed() {
+    const cards = await this.getAllCards();
+
+    for (const card of cards) {
+      const cardText = await card.getText();
+      const isDisplayed = await card.isDisplayed();
+
+      if (cardText.includes("Urgent") && isDisplayed) {
```

Comment:
does "Urgent" is text in title or it's label or some card option?
if text, try to make reusable functions(there is no need to implement now, do it in scope of test automation framework module task)


```
  async checkOnlyCardsContainsTextDisplayed(text) {
    const cards = await this.getAllCards();

    for (const card of cards) {
      const cardText = await card.getText();
      const isDisplayed = await card.isDisplayed();

      if (cardText.includes(text) && isDisplayed) {
              continue;
      } else if (isDisplayed) {
         throw new Error('Card with "${cardText}" text is Displayed')
      }
}
```
we should try to make our tests independent from testData changes(at least from simple changes, like card name or title name and etc), or at least reduce places where it's hardcoded

Reply:
for this test I needed a precondition, where cards should have the label "Urgent", so yes this is a label for cards